### PR TITLE
K8SPS-141: improve telemetry

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-141](https://jira.percona.com/browse/K8SPS-141) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPS-141

**DESCRIPTION**
---
**Problem:**
*We need to gather telemetry data about Operator installations and active clusters, the same way we do for other operators.*

**Solution:**
*Add new env vars `DISABLE_TELEMETRY` and `PERCONA_VS_FALLBACK_URI` to the operator. Send telemetry with the same logic we have in other operators*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Are E2E tests passing?
- [x] Are unit tests passing?
- [x] Are linting tests passing?
- [ ] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PS version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/gr-init-deploy.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/gr-scaling.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/config.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/sidecars.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/one-pod.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/limits.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/gr-demand-backup.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/haproxy.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/tls-cert-manager.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/gr-tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/users.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/async-ignore-annotations.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/045401f/gr-ignore-annotations.log) |
| We run 21 out of 21|

commit: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/045401f11966139b27c6a7761d298ef6c94ee81c
image: `perconalab/percona-server-mysql-operator:PR-291-045401f`

**Review by hors — APPROVED** [review]

File: controllers/perconaservermysql_controller.go [hunk 1]
```diff
@@ -297,21 +300,43 @@ func (r *PerconaServerMySQLReconciler) doReconcile(
...
+		}
+	}
+
+	if !versionUpgradeEnabled(cr) {
 		return nil
 	}
 
+	version, err := vs.GetVersion(ctx, cr, cr.Spec.UpgradeOptions.VersionServiceEndpoint, r.ServerVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to get versions, using the default ones")
```
Comment (inelpandzic) [inline controllers/perconaservermysql_controller.go:337]: In fact `"failed to get versions, using the default ones"` should be a log, and for the error just what failed. Something like this:
```
log.Info("failed to get versions, using the default ones")
return errors.Wrap(err, "failed to get versions")
```

File: controllers/perconaservermysql_controller.go [hunk 2]
```diff
@@ -356,8 +381,7 @@ func (r *PerconaServerMySQLReconciler) reconcileVersions(ctx context.Context, cr
 
 	err = r.Patch(ctx, cr.DeepCopy(), patch)
 	if err != nil {
-		log.Info("Failed to update CR, using the default version", "error", err)
-		return nil
+		return errors.Wrap(err, "failed to update CR, using the default version")
```
Comment (inelpandzic) [inline controllers/perconaservermysql_controller.go:384]: Same as for `failed to get versions, using the default ones` case.

Comment (pooknull) [reply controllers/perconaservermysql_controller.go]: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/cf68585fb494946f31a373103334d5c0c0a1a51f

File: controllers/perconaservermysql_controller.go (hunk 1 above)
Comment (pooknull) [reply controllers/perconaservermysql_controller.go]: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/cf68585fb494946f31a373103334d5c0c0a1a51f

File: controllers/perconaservermysql_controller.go [hunk 3]
```diff
@@ -260,8 +261,10 @@ func (r *PerconaServerMySQLReconciler) doReconcile(
 	ctx context.Context,
 	cr *apiv1alpha1.PerconaServerMySQL,
 ) error {
+	log := logf.FromContext(ctx).WithName("doReconcile")
+
 	if err := r.reconcileVersions(ctx, cr); err != nil {
-		return errors.Wrap(err, "reconcile versions")
+		log.Info("reconcileVersions", "error", err.Error())
```
[the remark this answers is not in the document]
Comment (pooknull) [reply controllers/perconaservermysql_controller.go]: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/cf68585fb494946f31a373103334d5c0c0a1a51f

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/gr-init-deploy.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/gr-scaling.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/config.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/semi-sync.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/gr-demand-backup.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/sidecars.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/one-pod.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/limits.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/haproxy.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/version-service.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/scaling.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/users.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/gr-tls-cert-manager.log) |
| async-ignore-annotations | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/async-ignore-annotations.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/23620c8/gr-ignore-annotations.log) |
| We run 21 out of 21|

commit: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/23620c8ee3d2e8ecda876ffd5805a58a60b10666
image: `perconalab/percona-server-mysql-operator:PR-291-23620c8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/config.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/one-pod.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/haproxy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/gr-demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/gr-init-deploy.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/gr-scaling.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/version-service.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/async-ignore-annotations.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/gr-ignore-annotations.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/tls-cert-manager.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/7cf9072/gr-tls-cert-manager.log) |
| We run 21 out of 21|

commit: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/7cf907255413f1f02d028f2d6a35878790664eb8
image: `perconalab/percona-server-mysql-operator:PR-291-7cf9072`

File: internal/testutil/versionservice.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+							ImagePath: "router-image",
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func FakeVersionService(addr string, port int, gwport int, unimplemented bool) *fakeVS {
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply internal/testutil/versionservice.go]: @pooknull wdyt about this?

**Review by hors — DISMISSED** [review]

**Review by inelpandzic — APPROVED** [review]
This is good for me. @pooknull just check https://github.com/percona/percona-server-mysql-operator/pull/291#discussion_r1118386725, if it is a big thing to do right now, we can refactor it later. Up to you, just note it so we don't forget it.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/gr-init-deploy.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/gr-scaling.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/config.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/semi-sync.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/one-pod.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/service-per-pod.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/gr-demand-backup.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/haproxy.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/tls-cert-manager.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/gr-tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/users.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/async-ignore-annotations.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/bc6a4e7/gr-ignore-annotations.log) |
| We run 21 out of 21|

commit: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/bc6a4e7858512798595390a43819296a73cccbb6
image: `perconalab/percona-server-mysql-operator:PR-291-bc6a4e7`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/gr-init-deploy.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/gr-scaling.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/monitoring.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/gr-demand-backup.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/limits.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/scaling.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/users.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/config-router.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/tls-cert-manager.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/async-ignore-annotations.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/gr-tls-cert-manager.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/gr-ignore-annotations.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/one-pod.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-291/3953d7b/haproxy.log) |
| We run 22 out of 22|

commit: https://github.com/percona/percona-server-mysql-operator/pull/291/commits/3953d7bdf86a955a98798af4a777745a0a6dd813
image: `perconalab/percona-server-mysql-operator:PR-291-3953d7b`

**Review by hors — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Pull request merged**